modules/barcode/src/detector/bardetect.cpp

```diff
@@ -0,0 +1,422 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Copyright (c) 2020-2021 darkliang wangberlinT Certseeds
+
+#include "../precomp.hpp"
+#include "bardetect.hpp"
+
+
+namespace cv {
+namespace barcode {
+static constexpr float PI = static_cast<float>(CV_PI);
+static constexpr float HALF_PI = static_cast<float>(CV_PI / 2);
+
+#define CALCULATE_SUM(ptr, result) \
+    ptr += left_col + integral_cols * top_row;\
+    top_left = static_cast<float>(*(ptr));\
+    (ptr) += right_col - left_col;\
+    top_right = static_cast<float>(*(ptr));\
+    (ptr) += (bottom_row - top_row) * integral_cols;\
+    bottom_right = static_cast<float>(*(ptr));\
+    (ptr) -= right_col - left_col;\
+    bottom_left = static_cast<float>(*(ptr));\
+    (ptr) -= (bottom_row - top_row) * integral_cols;\
+    (ptr) -= left_col + integral_cols * top_row;\
```

Comment:
> `(ptr) += ...`
> `(ptr) -= ...`

It doesn't make sense to modify `ptr` itself.
Modern compilers are smart enough to detect and optimize common sub-expressions.

(also such modification may block many optimizations in compilers, e.g `ptr` is a class field, so it must be update with all intermediate results)